crates/isograph_schema/src/validate_schema.rs

```diff
@@ -440,6 +440,31 @@ fn validate_and_transform_client_fields(
     }))
 }
 
+fn validate_all_variables_are_used(
+    variable_definitions: Vec<WithSpan<UnvalidatedVariableDefinition>>,
+    used_variables: BTreeSet<VariableName>,
+) -> ValidateSelectionsResult<()> {
+    let unused_variables: Vec<_> = variable_definitions
+        .iter()
+        .filter_map(|variable| {
+            let is_used = used_variables.contains(&variable.item.name.item);
+
+            if !is_used {
+                return Some(variable.clone());
+            }
+            None
+        })
+        .collect();
+
+    if !unused_variables.is_empty() {
+        return Err(WithLocation::new(
+            ValidateSelectionsError::UnusedVariables { unused_variables },
+            Location::generated(),
```

Comment:
I'm surprised that `ClientField`'s do not have location information for the name.

If you want to do that, you can change the `ClientField`'s `name` field to be `WithLocation<SelectableFieldName>`.

Happy to also do `Location::generated` here in the mean time. (I think I want to rethink a few things about how we store locations.) If we do that, we might want to add the client field's parent type and name and render it as `Type.name` in the error message

Reply:
I think it's good enough for now, this is easy to tweak later, and probably will with the work on vs code extension